### PR TITLE
run tee-worker with remote attestation

### PR DESCRIPTION
**Pull request by zTgx**
run `integritee-service` without flag `--skip-ra` to enable ra. 
~~and with the worker start command likes~~

* enable ra
~~support linkable and unlikable quote type with flag `--linkable`~~
 
~~./integritee-service \
	--mu-ra-external-address "${worker_endpoint}" --mu-ra-port 3443 --untrusted-http-port 4545 --ws-external \
	--trusted-external-address "wss://${worker_endpoint}" --trusted-worker-port 2000 \
	--untrusted-external-address "ws://${worker_endpoint}" --untrusted-worker-port 2001 \
	--node-url "ws://${node_endpoint}" --node-port "${node_port}" \
	run --linkable --dev >"${ROOTDIR}"/log/worker1.log 2>&1 &~~

- [x] fix `get_sigrl_from_intel` 's param `gid` encoding format
~~fix enclave crash
add ra_tests~~

---

As integritee doc said, `We use unlinkable quotes in our code with enhanced privacy.`, and we don't necessarily need `linkable` quote either. so i revert `linkable` option.

Test steps:
1. replace `spid.txt`/ `key.txt` with `unlinkable` quote
2. run enclave test alone with command `test`
```shell
RUST_LOG="info,ws=warn,sp_io=error,substrate_api_client=warn,jsonrpsee_ws_client=warn,jsonrpsee_ws_server=warn, enclave_runtime=warn, integritee_service=info,ita_stf=debug,its_rpc_handler=warn,itc_rpc_client=warn,its_consensus_common=warn,its_state=warn,its_consensus_aura=warn,aura*=warn,its_consensus_slots=warn" \
	./integritee-service \
	--mu-ra-external-address "${worker_endpoint}" --mu-ra-port 3443 --untrusted-http-port 4545 --ws-external \
	--trusted-external-address "wss://${worker_endpoint}" --trusted-worker-port 2000 \
	--untrusted-external-address "ws://${worker_endpoint}" --untrusted-worker-port 2001 \
	--node-url "ws://${node_endpoint}" --node-port "${node_port}" \
	test -a >"${ROOTDIR}"/log/worker1.log 2>&1 &
```
3. using command to run `integritee-service` without `--skip-ra`
```shell
./integritee-service \
	--mu-ra-external-address "${worker_endpoint}" --mu-ra-port 3443 --untrusted-http-port 4545 --ws-external \
	--trusted-external-address "wss://${worker_endpoint}" --trusted-worker-port 2000 \
	--untrusted-external-address "ws://${worker_endpoint}" --untrusted-worker-port 2001 \
	--node-url "ws://${node_endpoint}" --node-port "${node_port}" \
	run --dev >"${ROOTDIR}"/log/worker1.log 2>&1 &
```

### CODE REVIEW

File: tee-worker/service/src/cli.yml [hunk 1]
```diff
@@ -122,6 +126,11 @@ subcommands:
                   long: skip-ra
                   short: s
                   help: skip remote attestation. Set this flag if running enclave in SW mode
+            - linkable:
+                long: linkable
+                short: l
+                help: Set this flag if this SPID is associated with linkable quotes.
+
```
Comment (BillyWooo) [inline tee-worker/service/src/cli.yml]: duplicate code block.

Comment (zTgx) [reply tee-worker/service/src/cli.yml]: those codes are no dup, this `linkable` arg is the args of `request-state` command, not for `run`.

Comment (BillyWooo) [reply tee-worker/service/src/cli.yml]: oh, yes. My mistake.

**Comment by Kailai-Wang** [comment]
I think the code itself is OK, but one thing to clarify:
AFAIU this PR adds an option to enable a `linkable` quote, with it defaults to be false. But in integritee's code I see comment like:
```
// Our certificate is unlikable. 
```

so have you tested it's supported or not? And why do we absolutely need a linkable quote?

**Comment by zTgx** [comment]
> I think the code itself is OK, but one thing to clarify: AFAIU this PR adds an option to enable a `linkable` quote, with it defaults to be false. But in integritee's code I see comment like:
> 
> ```
> // Our certificate is unlikable. 
> ```
> 
> so have you tested it's supported or not? And why do we absolutely need a linkable quote?

yes, I've tested it.
I'm using a `linkable` quote to run `integritee-service` at first,  but then it crashed. so I think it's more flexible to support linkable and unlinkable both. 
I read the `integritee` docs last night, [here](https://docs.integritee.network/5-nodes-and-infrastructure/5.2-how-to-set-up-and-run-a-sidechain-or-tocw-node/5.2.2-intel-sgx-licences), and rethinking this option, now I think it seems that we don't necessarily need to support linkable quote.
So, I need to revert this `linkable` option mabye.

**Review by jingleizhang — APPROVED** [review]
LGTM

**Comment by Kailai-Wang** [comment]
Thanks, shall we submit an upstream PR as well? It looks like a bug of integritee?

**Comment by jingleizhang** [comment]
It's good to submit a PR to integritee as well.

**Comment by Kailai-Wang** [comment]
> It's good to submit a PR to integritee as well.

Yes please go ahead and create an upstream PR @zTgx 
please attach our issue/PR as links

**Comment by zTgx** [comment]
> > It's good to submit a PR to integritee as well.
> 
> Yes please go ahead and create an upstream PR @zTgx please attach our issue/PR as links

ok, will do.

submitted
https://github.com/integritee-network/worker/pull/1118

**Comment by jingleizhang** [comment]
The remote attestation on integritee was not broken. Close this issue.

**Pull request closed without merging**